<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="noindex, nofollow">
<title>10X_scCancer | My New Hugo Site</title>
<meta name="keywords" content="">
<meta name="description" content="本篇主要记录基于10X cellranger之后的数据进行简单的初步处理。以及简单的报告，该部分内容基于 scCancer2
如何安装scCancer2 如何使用scCancer2进行分析 工具介绍 以下来自于 scCancer2的github介绍
We updated our R toolkit, scCancer, based on massive single-cell transcriptome and spatial transcriptome data.
Cell subtype annotation and cross-dataset label similarity: Our analysis mainly focused on cell subtype annotation by training multiple lightweight machine-learning models on scRNA-seq data. We proposed a method for quantitatively evaluating the similarity of cell subtype labels originating from different published datasets. We fully preserved the original labeling in cell atlases and analyzed the relationship between cell subtypes across datasets.">
<meta name="author" content="">
<link rel="canonical" href="https://gganakin.github.io/posts/10x_sccancer/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.a558ac8f16837d1ca9780489fb47c199b58071c5c356b9c84d8acae50728cd7b.css" integrity="sha256-pVisjxaDfRypeASJ&#43;0fBmbWAccXDVrnITYrK5QcozXs=" rel="preload stylesheet" as="style">
<link rel="icon" href="https://gganakin.github.io/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://gganakin.github.io/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://gganakin.github.io/favicon-32x32.png">
<link rel="apple-touch-icon" href="https://gganakin.github.io/apple-touch-icon.png">
<link rel="mask-icon" href="https://gganakin.github.io/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="https://gganakin.github.io/posts/10x_sccancer/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="https://gganakin.github.io/" accesskey="h" title="My New Hugo Site (Alt + H)">My New Hugo Site</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    
    <h1 class="post-title entry-hint-parent">
      10X_scCancer
      <span class="entry-hint" title="Draft">
        <svg xmlns="http://www.w3.org/2000/svg" height="35" viewBox="0 -960 960 960" fill="currentColor">
          <path
            d="M160-410v-60h300v60H160Zm0-165v-60h470v60H160Zm0-165v-60h470v60H160Zm360 580v-123l221-220q9-9 20-13t22-4q12 0 23 4.5t20 13.5l37 37q9 9 13 20t4 22q0 11-4.5 22.5T862.09-380L643-160H520Zm300-263-37-37 37 37ZM580-220h38l121-122-18-19-19-18-122 121v38Zm141-141-19-18 37 37-18-19Z" />
        </svg>
      </span>
    </h1>
    <div class="post-meta"><span title='2024-10-16 11:37:57 +0800 CST'>October 16, 2024</span>

</div>
  </header> 
  <div class="post-content"><p>本篇主要记录基于10X cellranger之后的数据进行简单的初步处理。以及简单的报告，该部分内容基于 <a href="https://github.com/czythu/scCancer">scCancer2</a></p>
<ul>
<li>如何安装scCancer2</li>
<li>如何使用scCancer2进行分析</li>
</ul>
<h1 id="工具介绍">工具介绍<a hidden class="anchor" aria-hidden="true" href="#工具介绍">#</a></h1>
<p>以下来自于 scCancer2的github介绍</p>
<p><a href="https://doi.org/10.5281/zenodo.10477297"><img loading="lazy" src="https://zenodo.org/badge/DOI/10.5281/zenodo.10477297.svg" alt="DOI"  />
</a></p>
<p>We updated our R toolkit, <code>scCancer</code>, based on massive single-cell transcriptome and spatial transcriptome data.</p>
<ol>
<li>
<p><code>Cell subtype annotation</code> and <code>cross-dataset label similarity</code>: Our analysis mainly focused on cell subtype annotation by training multiple lightweight machine-learning models on scRNA-seq data. We proposed a method for quantitatively evaluating the similarity of cell subtype labels originating from different published datasets. We fully preserved the original labeling in cell atlases and analyzed the relationship between cell subtypes across datasets.</p>
</li>
<li>
<p><code>Malignant cell identification</code>: We constructed a reference dataset combining scRNA-seq and bulk RNA-seq data across multiple cancer types to identify the malignant cell in TME. We trained a model to identify malignant cells with high generalization ability and computational efficiency.</p>
</li>
<li>
<p><code>Spatial transcriptome analysis</code>: Finally, we integrated a spatial transcriptome analysis pipeline. It enables us to analyze TME from a spatial perspective systematically and automatically.</p>
</li>
</ol>
<p>With <code>scCancer2</code>, researchers can understand the composition of the TME more accurately from multiple dimensions.</p>
<p><img loading="lazy" src="Overview.png" alt="image"  />
</p>
<h1 id="工具安装">工具安装<a hidden class="anchor" aria-hidden="true" href="#工具安装">#</a></h1>
<p>目前该工具依赖于 Seurat-4.3.0。其余的按照github页面的安装指引即可。(还是挺折腾的，建议单独开一个虚拟环境安装QAQ)
harmony的版本不能低了</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>micromamba install conda-forge::r-harmony
</span></span></code></pre></div><p>值得注意的是在该工具存在一个bug(在 R 4.3.2 中)。引起错误的原因出现在 <a href="https://github.com/czythu/scCancer/blob/master/R/scStatistics.R">scStatistics.R</a>的第257行。</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>tmp.df <span style="color:#f92672">&lt;-</span> cell.manifest<span style="color:#a6e22e">[order</span>(cell.manifest[<span style="color:#e6db74">&#34;nUMI&#34;</span>], decreasing <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>),]
</span></span></code></pre></div><p>应该修改为</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>tmp.df <span style="color:#f92672">&lt;-</span> cell.manifest<span style="color:#a6e22e">[order</span>(cell.manifest[,<span style="color:#e6db74">&#34;nUMI&#34;</span>], decreasing <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>),]
</span></span><span style="display:flex;"><span><span style="color:#75715e">## 或</span>
</span></span><span style="display:flex;"><span>tmp.df <span style="color:#f92672">&lt;-</span> cell.manifest<span style="color:#a6e22e">[order</span>(cell.manifest<span style="color:#f92672">$</span>nUMI, decreasing <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>),]
</span></span></code></pre></div><p>需要解压源码包修改其中的  <em>R/scStatistics.R</em> 中的代码重新安装。也可以用我的github下面fork的版本。</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>devtools<span style="color:#f92672">::</span><span style="color:#a6e22e">install_github</span>(<span style="color:#e6db74">&#34;https://github.com/gganakin/scCancer2&#34;</span>)
</span></span></code></pre></div><h1 id="使用方法">使用方法<a hidden class="anchor" aria-hidden="true" href="#使用方法">#</a></h1>
<p>具体可以参考旧版本的<a href="https://github.com/wguo-research/scCancer/wiki/3.-Quick-start">使用说明</a>，使用方法没有变化。
加载安装好的R包</p>
<pre tabindex="0"><code>library(scCancer)
</code></pre><h2 id="1-单样本分析">1. 单样本分析<a hidden class="anchor" aria-hidden="true" href="#1-单样本分析">#</a></h2>
<h3 id="scstatistics">ScStatistics<a hidden class="anchor" aria-hidden="true" href="#scstatistics">#</a></h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>dataPath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/cellranger/run_sample/outs/&#34;</span>     <span style="color:#75715e"># 需要指定到cellranger结果文件中的 outs/ 文件夹</span>
</span></span><span style="display:flex;"><span>savePath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/results/Sample/ScStat&#34;</span>           <span style="color:#75715e"># 存放本次分析结果路径，由于该工具分为 statistic 和 annotation 两个步骤，所以可以指定不同的路径</span>
</span></span><span style="display:flex;"><span>sampleName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;Sample&#34;</span>              <span style="color:#75715e"># 样本名称</span>
</span></span><span style="display:flex;"><span>authorName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;author&#34;</span>              <span style="color:#75715e"># 出现在报告文件中的名字</span>
</span></span><span style="display:flex;"><span>species <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;human&#34;</span>                  <span style="color:#75715e"># 指定物种，目前只有人(human) 和小鼠(mouse)</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Run scStatistics</span>
</span></span><span style="display:flex;"><span>stat.results <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">runScStatistics</span>(
</span></span><span style="display:flex;"><span>    dataPath <span style="color:#f92672">=</span> dataPath,
</span></span><span style="display:flex;"><span>    savePath <span style="color:#f92672">=</span> savePath,
</span></span><span style="display:flex;"><span>    sampleName <span style="color:#f92672">=</span> sampleName,
</span></span><span style="display:flex;"><span>    authorName <span style="color:#f92672">=</span> authorName,
</span></span><span style="display:flex;"><span>    species <span style="color:#f92672">=</span> species
</span></span><span style="display:flex;"><span>)
</span></span></code></pre></div><p>会在 你指定的 <em>savePath</em> 也就是 <em>/path/to/results/Sample/ScStat</em> 下生成这些文件。
文件夹下的report-scStat.html 就是总览文件</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-text" data-lang="text"><span style="display:flex;"><span>├── cellManifest-all.txt
</span></span><span style="display:flex;"><span>├── cell.QC.thres.txt
</span></span><span style="display:flex;"><span>├── figures
</span></span><span style="display:flex;"><span>├── geneManifest.txt
</span></span><span style="display:flex;"><span>├── report-cellRanger.html
</span></span><span style="display:flex;"><span>├── report-figures
</span></span><span style="display:flex;"><span>├── report-scStat.html
</span></span><span style="display:flex;"><span>├── report-scStat.md
</span></span><span style="display:flex;"><span>└── scStatistics-results.RDS
</span></span></code></pre></div><ul>
<li>cellManifest-all.txt : The statistical results for all droplets.</li>
<li>cell.QC.thres.txt : The suggested thresholds to filter poor-quality cells.</li>
<li>figures/ : All figures generated during this module.</li>
<li>geneManifest.txt : The statistical results for genes.</li>
<li>report-cellRanger.html : The summary report generated by Cell Ranger.</li>
<li>report-figures/ : All figures presented in the HTML report.</li>
<li>report-scStat.html : A HTML report containing all results.</li>
<li>report-scStat.md : A markdown report.</li>
<li>scStatistics-results.RDS : A RDS file containing all results.</li>
</ul>
<h3 id="scannotation">ScAnnotation<a hidden class="anchor" aria-hidden="true" href="#scannotation">#</a></h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>dataPath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/cellranger/run_sample/outs/&#34;</span>  <span style="color:#75715e">## cellranger 输出路径 </span>
</span></span><span style="display:flex;"><span>savePath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/results/Sample/ScAnno&#34;</span>        <span style="color:#75715e">## 结果存放路径</span>
</span></span><span style="display:flex;"><span>sampleName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;Sample&#34;</span>                              <span style="color:#75715e">## 样本名称</span>
</span></span><span style="display:flex;"><span>authorName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;author&#34;</span>                              <span style="color:#75715e">## 分析者</span>
</span></span><span style="display:flex;"><span>species <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;human&#34;</span>                                  <span style="color:#75715e">## 物种</span>
</span></span><span style="display:flex;"><span>statPath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/results/Sample/ScStat&#34;</span>        <span style="color:#75715e">## scStatistics 输出路径</span>
</span></span><span style="display:flex;"><span>geneSet.method <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;average&#34;</span>                         <span style="color:#75715e">## 细胞类型定义方法 average or GSVA</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Run scAnnotation</span>
</span></span><span style="display:flex;"><span>anno.results <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">runScAnnotation</span>(
</span></span><span style="display:flex;"><span>    dataPath <span style="color:#f92672">=</span> dataPath,
</span></span><span style="display:flex;"><span>    statPath <span style="color:#f92672">=</span> statPath,
</span></span><span style="display:flex;"><span>    savePath <span style="color:#f92672">=</span> savePath,
</span></span><span style="display:flex;"><span>    authorName <span style="color:#f92672">=</span> authorName,
</span></span><span style="display:flex;"><span>    sampleName <span style="color:#f92672">=</span> sampleName,
</span></span><span style="display:flex;"><span>    geneSet.method <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;average&#34;</span>       <span style="color:#75715e"># or &#34;GSVA&#34;</span>
</span></span><span style="display:flex;"><span>)
</span></span></code></pre></div><p>会在 你指定的 <em>savePath</em> 也就是 <em>/path/to/results/Sample/ScAnno</em> 下生成这些文件。</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-text" data-lang="text"><span style="display:flex;"><span>├── cellAnnotation.txt
</span></span><span style="display:flex;"><span>├── cellSubtypeAnno
</span></span><span style="display:flex;"><span>├── DE-Genes.RDS
</span></span><span style="display:flex;"><span>├── diff.expr.genes
</span></span><span style="display:flex;"><span>├── expr.programs
</span></span><span style="display:flex;"><span>├── expr.RDS
</span></span><span style="display:flex;"><span>├── figures
</span></span><span style="display:flex;"><span>├── geneManifest.txt
</span></span><span style="display:flex;"><span>├── InteractionScore.txt
</span></span><span style="display:flex;"><span>├── malignancy-inferCNV
</span></span><span style="display:flex;"><span>├── report-figures
</span></span><span style="display:flex;"><span>├── report-scAnno.html
</span></span><span style="display:flex;"><span>├── report-scAnno.md
</span></span><span style="display:flex;"><span>├── rough-cell-annotation.RDS
</span></span><span style="display:flex;"><span>└── rough-labels.RDS
</span></span></code></pre></div><ul>
<li>cellAnnotation.txt : The annotation results for each cells.</li>
<li>cellSubtypeAnno : The annotation results for each cells.</li>
<li>DE-Genes.RDS : Differentially expressed genes information for all clusters.</li>
<li>diff.expr.genes/ : Differentially expressed genes information for all clusters.</li>
<li>expr.programs/ : All results of expression programs identification.</li>
<li>expr.RDS : A Seurat object.</li>
<li>malignancy-inferCNV : The results of malignancy estimation.</li>
<li>report-figures/ : All figures presented in the HTML report.</li>
<li>report-scAnno.html : A HTML report containing all results.</li>
<li>report-scAnno.md : A markdown report.</li>
<li>rough-cell-annotation.RDS : A RDS file containing cell type.</li>
<li>rough-labels.RDS : A RDS file containing cell label.</li>
</ul>
<p>一般也就用这里的expr.RDS文件。以及查看html报告</p>
<h2 id="2-多样本分析">2. 多样本分析<a hidden class="anchor" aria-hidden="true" href="#2-多样本分析">#</a></h2>
<h3 id="sccombination">ScCombination<a hidden class="anchor" aria-hidden="true" href="#sccombination">#</a></h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-R" data-lang="R"><span style="display:flex;"><span>single.savePaths <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;/path/to/results/Sample1/ScAnno&#34;</span>,
</span></span><span style="display:flex;"><span>                     <span style="color:#e6db74">&#34;/path/to/results/Sample2/ScAnno&#34;</span>,
</span></span><span style="display:flex;"><span>                     <span style="color:#e6db74">&#34;/path/to/results/Sample3/ScAnno&#34;</span>,
</span></span><span style="display:flex;"><span>                     <span style="color:#e6db74">&#34;/path/to/results/Sample4/ScAnno&#34;</span>,
</span></span><span style="display:flex;"><span>                     <span style="color:#e6db74">&#34;/path/to/results/Sample5/ScAnno&#34;</span>,
</span></span><span style="display:flex;"><span>                     <span style="color:#e6db74">&#34;/path/to/results/Sample6/ScAnno&#34;</span>)
</span></span><span style="display:flex;"><span>sampleNames <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;Sample1&#34;</span>, <span style="color:#e6db74">&#34;Sample2&#34;</span>, <span style="color:#e6db74">&#34;Sample3&#34;</span>, <span style="color:#e6db74">&#34;Sample4&#34;</span>, <span style="color:#e6db74">&#34;Sample5&#34;</span>, <span style="color:#e6db74">&#34;Sample6&#34;</span>) <span style="color:#75715e">## 一定要和single.savePath一一对应</span>
</span></span><span style="display:flex;"><span>savePath <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;/path/to/results/Combination&#34;</span>
</span></span><span style="display:flex;"><span>authorName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;author&#34;</span>
</span></span><span style="display:flex;"><span>combName <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;Combination&#34;</span>
</span></span><span style="display:flex;"><span>combine.method <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;Harmony&#34;</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Run scCombination</span>
</span></span><span style="display:flex;"><span>comb.results <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">runScCombination</span>(
</span></span><span style="display:flex;"><span>    single.savePaths <span style="color:#f92672">=</span> single.savePaths, 
</span></span><span style="display:flex;"><span>    sampleNames <span style="color:#f92672">=</span> sampleNames, 
</span></span><span style="display:flex;"><span>    savePath <span style="color:#f92672">=</span> savePath, 
</span></span><span style="display:flex;"><span>    combName <span style="color:#f92672">=</span> combName,
</span></span><span style="display:flex;"><span>    authorName <span style="color:#f92672">=</span> authorName,
</span></span><span style="display:flex;"><span>    comb.method <span style="color:#f92672">=</span> comb.method
</span></span><span style="display:flex;"><span>)
</span></span></code></pre></div><p>会在 <em>savePath</em> 也就是 <em>/path/to/results/Combination</em> 下生成这些文件。</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-text" data-lang="text"><span style="display:flex;"><span>├── cellAnnotation.txt
</span></span><span style="display:flex;"><span>├── DE-Genes.RDS
</span></span><span style="display:flex;"><span>├── diff.expr.genes
</span></span><span style="display:flex;"><span>├── expr.programs
</span></span><span style="display:flex;"><span>├── expr.RDS
</span></span><span style="display:flex;"><span>├── figures
</span></span><span style="display:flex;"><span>├── report-figures
</span></span><span style="display:flex;"><span>├── report-scAnnoComb.html
</span></span><span style="display:flex;"><span>└── report-scAnnoComb.md
</span></span></code></pre></div><ul>
<li>report-scAnnoComb.html : A HTML report containing all results.</li>
<li>report-scAnnoComb.md : A markdown report.</li>
<li>figures/ : All figures generated during this step.</li>
<li>report-figures/ : All figures presented in the HTML report.</li>
<li>expr.RDS : A Seurat object.</li>
<li>diff.expr.genes/ : Differentially expressed genes information for all clusters.</li>
<li>cellAnnotation.txt : The annotation results for each cells.</li>
<li>expr.programs/ : All results of expression programs identification.</li>
<li>(anchors.RDS : The anchors used for batch correction of &ldquo;NormalMNN&rdquo; or &ldquo;SeuratMNN&rdquo;.)</li>
</ul>


  </div>

  <footer class="post-footer">
    <ul class="post-tags">
    </ul>
  </footer>
</article>
    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="https://gganakin.github.io/">My New Hugo Site</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
